extensions-core/kafka-indexing-service/src/main/java/io/druid/indexing/kafka/supervisor/KafkaSupervisor.java

```diff
@@ -1116,22 +1152,20 @@ private void killTask(final String id)
 
   private DateTime getTaskStartTime(final String id)
   {
-    DateTime startTime = null;
-    TaskLocation taskLocation = getTaskLocation(id);
-    if (taskLocation != null) {
-      startTime = taskClient.getStartTime(taskLocation, id);
+    if (taskInfoProvider.getTaskLocation(id) != null) {
```

Comment:
is there a potential race here? like you could have the task present here but vanish before `getStartTime`?